### ISSUE TITLE
result counters on search pages reset after page 1

### ISSUE THREAD

**Issue by thatbudakguy** [Princeton-CDH/mep-django#447]
## notes for testing
Check both members and books search/browse pages. Click to the next page of results and confirm that the numbers start at 101, 201, etc instead of starting at every time.

---
this issue occurs when asynchronously loading new results on search pages. because the result counter styles need to be updated to correctly set the number on each result, we use an inline style in the partial that is requested:
```css
<style nonce="2726c7f26c">
ol { counter-reset: results 0; }
</style>
```

however, this is blocked by CSP when requested over ajax, which results in the result numbers being wrong because the style isn't applied.

see #396 for previous discussion.

**Comment by thatbudakguy** [comment]
enabled `unsafe-inline` for the `style-src` directive, but the problem wasn't resolved. it seems this is almost certainly due to the fact that this is loaded via ajax.

**Comment by rlskoeser** [comment]
@thatbudakguy ok, so I messed with this long enough to a) remember what we did here and why and b) finally understand what is going on. We have a nonce set on the inline style, but the nonce for the ajax request is _different_ than the original page request, so it's not allowed.   What's more, `unsafe-inline` doesn't take effect if you have nonces turned on for style src. 

I think the two options are:
1. Remove `'style-src'` from our **CSP_INCLUDE_NONCE_IN** config and remove the nonce attribute from the inline style
2. Update the nonce in the typescript code before inserting the html

Option 1 seems easier; option 2 probably isn't that hard, but I can't tell how much more secure it is (unless we check the nonce in the code before replacing it, which makes it more complicated to implement, but safer).

Which option do you prefer?

**Comment by clmahoney** [comment]
- YES, I checked on the member page and clicked through the pages (NEXT) a couple of times. Every time I went to a new section of names, the top numbers reflected that I had moved through many pages already. For example, clicking through twice, the first name (Baker, Dan) was 201, not 1. Clicking next again, the first name was now (Bauerberg, Linda) was 301. When I clicked "previous" the numbers remained where they should (i.e. not revert back to 1 at the top of each page)

- YES, on the book search I browsed by clicking "next," and the numbers reflected the amount of book names I had gone through. The first click got me to 101 at the top, then the next click got me to 201. Numbers were good when I clicked previous too!
